src/Transformation/Transformation.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Transformation;
+
+/**
+ * A transformation is a function from one datatype to another.
+ *
+ * It MUST NOT perform any sideeffects, i.e. it must be morally impossible to observe
+ * how often the transformation was actually performed. It MUST NOT touch the providedi
+ * value, i.e. it is allowed to create new values but not to modify existing values.i
+ * This would be an observable sideeffect.  
+ */
+interface Transformation {
+	/**
+	 * Perform the transformation.
+	 *
+	 * @throws \InvalidArgumentException  if the argument could not be transformed
+	 * @param  mixed  $from
+	 * @return mixed
+	 */
+	public function transform($from);
```

Comment:
So it's the way we would like to encourage to use? maybe we should write it down here in the interface and give the reason why (performance)